index.ts

```diff
@@ -9,97 +9,52 @@ export async function main() {
-  // Which key the upload is going to
-  fileKey: string;
-
-  // Which URL to POST an upload to
-  url: string;
-
-  // Which fields to include in the POST
-  fields: { [key: string]: string };
-};
-
-async function createSignedUpload(token: string): Promise<SignedUploadResponse> {
-  const response = await fetch('https://api.stainlessapi.com/api/spec/upload', {
-    method: 'POST',
-    headers: {
-      Authorization: `Bearer ${token}`,
-      'Content-Type': 'application/json',
-    },
-  });
-
+  info('Uploading spec and config files...');
+  const response = await uploadSpecAndConfig(inputPath, configPath, stainless_api_key);
   if (!response.ok) {
-    const errorMsg = `Failed to create a signed upload: ${response.statusText} ${response.text}`;
+    const errorMsg = `Failed to upload spec or config file: ${response.statusText} ${response.text}`;
     error(errorMsg);
     throw Error(errorMsg);
   }
 
-  return response.json() as unknown as SignedUploadResponse;
+  if (outputPath) {
+    const decoratedSpec = response.text();
+    writeFile(outputPath, decoratedSpec);
+    info('Wrote decorated spec to', outputPath);
+  }
 }
 
-async function uploadSpec(specPath: string, upload: SignedUploadResponse) {
-  const { fields, url } = upload;
+async function uploadSpecAndConfig(specPath: string, configPath: string, token: string) {
   const formData = new FormData();
 
-  // Add the required fields for S3 upload
-  Object.entries(fields).forEach(([key, value]) => {
-    formData.append(key, value as string);
+  // append a spec file
+  const specStats = fs.statSync(specPath);
+  formData.append('oasSpec', fs.createReadStream(specPath), {
+    contentType: 'text/plain',
+    knownLength: specStats.size,
   });
 
-  // Attach the actual spec file
-  const stats = fs.statSync(specPath);
-  formData.append('file', fs.createReadStream(specPath), { knownLength: stats.size });
-
-  const response = await fetch(url, {
-    method: 'POST',
-    body: formData,
-  });
-
-  if (!response.ok) {
-    const errorMsg = `Failed to upload spec: ${response.statusText} ${response.text}`;
-    error(errorMsg);
-    throw Error(errorMsg);
+  // append a config file, if present
+  if (configPath) {
+    const configStats = fs.statSync(configPath);
+    formData.append('stainlessConfig', fs.createReadStream(specPath), {
+      contentType: 'text/plain',
+      knownLength: configStats.size,
+    });
   }
-}
 
-async function decorateSpec(specPath: string, token: string): Promise<string> {
-  info('Getting a signed upload URL...');
-  const signedUpload = await createSignedUpload(token);
-
-  info('Uploading the spec file...');
-  await uploadSpec(specPath, signedUpload);
-
-  info('Decorating spec...');
   const response = await fetch('https://api.stainlessapi.com/api/spec', {
     method: 'POST',
+    body: formData,
     headers: {
       Authorization: `Bearer ${token}`,
-      'Content-Type': 'application/json',
+      'Content-Type': 'multipart/form-data',
```

Comment:
doesn't have to happen in this PR but I think we should send a header indicating that the upload came from the github action so we can monitor it and provide better logging

Reply:
Good idea 👍

Thanks for the review too :)

Reply:
`X-GitHub-Action: true`, or set a specific user agent? What do you think?

Reply:
I went with `"X-Github-Action": "stainless-api/upload-openapi-spec-action"`, but maybe the user agent header would make sense here 🤷